### ISSUE TITLE
change the API to be more high-level?

### ISSUE THREAD

**Issue by neuhaus** [4ley/nodejs-bitcoin-de-api#6]
Right now it's a pretty low-level library with commands such as

    bitcoinde.get('orders', { type: 'sell' })

What are your thoughts on providing more high-level functions such as

    bitcoinde.getOrderbook()

? That function could retrieve the `buy` and `sell` orders in parallel and return them as one. That's just one example.

**Comment by 4ley** [comment]
Sounds good. Let's add a parameter to selectively choose order type or none for all orders. What do you think?

**Comment by neuhaus** [comment]
Yes that's probably necessary due to the filtering that the API offers. Also there all those other methods in their API as well.

**Comment by 4ley** [comment]
True.

What do you think about adding sockets support?

**Comment by neuhaus** [comment]
Would be a great addition and some work...

**Comment by 4ley** [comment]
Agree

**Comment by neuhaus** [comment]
Or adding support to ccxt...

**Comment by 4ley** [comment]
What do you mean by that? How would this look like?

**Comment by neuhaus** [comment]
There is a library on npm called ccxt that supports many marketplaces.
bitcoin.de has quite a few quirks because it's not a real marketplace but I guess it could be supported.
